src/server/project.ts

```diff
@@ -1768,6 +1768,11 @@ namespace ts.server {
         getIncompleteCompletionsCache() {
             return this.projectService.getIncompleteCompletionsCache();
         }
+
+        /*@internal*/
+        getProgramForReferencedProject(configFileName: string) {
+            return this.projectService.findConfiguredProjectByProjectName(toNormalizedPath(configFileName))?.getCurrentProgram();

```

Comment:
`findConfiguredProjectByProjectName` simply looks up from a map, no loading necessary. I had forgotten about lazy project loading though—it doesn’t have an impact in the tests, but I need to develop an understanding of when/if this might not work as I intend.

Reply:
Yeah, I was more worried about the `getCurrentProgram` call.  I thought you could have a project that didn't (yet) have a program.

Reply:
`getCurrentProgram` returns `undefined` if there’s currently no program.